File: ejemplos_ethan/src/interfaces/Pato.java
package interfaces;

public class Pato extends Animal  implements Volador{

    public Pato(int age) {
        super(age);
    }

    @Override
    public void hacerRuido() {
        System.out.println("ruido");
    }

    @Override
    public void volar() {
        System.out.println("vuela");
    }
}

File: ClasesAbstractas/ejemplos_erick/src/animals/Pato.ts
import {Animal} from "./Animal.ts";
import {Volar} from "../skills/Volar.ts";
import {Nadar} from "../skills/Nadar.ts";

export class Pato extends Animal implements Volar, Nadar{
    constructor(name: string) {
        super(name);
    }

    hacerSonido(): void {
        console.log(`Cuak`);
    }


    volar(): void {
        console.log("El pato vuela");
    }

    nadar(): void {
        console.log("El pato nada");
    }
}

